src/client/src/system/openFin/openFin.js

```diff
@@ -84,8 +85,52 @@ export default class OpenFin {
         return disposables;
       });
   }
-
+  
   get _currentWindow() {
     return fin.desktop.Window.getCurrent();
   }
+  
+  viewChartIQ(symbol): void {
+    this.getChartIQInstance(symbol, this._refreshChartIQ, this._startChartIQ);
+  }
+
+  getChartIQInstance(symbol, callback, errorHandler):fin.desktop.Application{
```

Comment:
we should wrap this up in RX:

```
getChartIQInstance(symbol) { // note removal of callbacks callback and errorHandler
 return Rx.Observable.create(observer => {
  let chartIqApp = _.find(apps, ((app) => {
    return app.isRunning && app.uuid === chartIqAppId;
  }));
  if(chartIqApp) {
    observer.onNext(symbol);
    observer.onCompleted();
  } else {
    o.onError(symbol);
  } 
 });
}

```

Then clean up the calling code so we don't have to use 'bind'. Additionally, we can signal via this stream when it completes so the calling code doesn't have to poll.

Happy to talk you through it